renku/core/util/git.py

```diff
@@ -545,8 +546,11 @@ def push_changes(repository: "Repository", remote: Optional[str] = None, reset:
 
         if merge_conflict or push_failed:
             # NOTE: Push to a new remote branch and reset the cache.
+            last_short_sha = repository.head.commit.hexsha[0:8]
             old_active_branch = repository.active_branch
-            pushed_branch = uuid4().hex
+            user = get_git_user(repository)
+            email = getattr(user, "email", "unknown")
+            pushed_branch = f"{PRETECTED_BRANCH_PREFIX}/{email}/{old_active_branch}/{last_short_sha}"
```

Comment:
I think we should check that the branch name length does not get too long. According to https://stackoverflow.com/questions/60045157/what-is-the-maximum-length-of-a-github-branch-name it's anywhere between 60 and 250 characters. We probably want to shorten `old_active_branch` down to a minimum of 6 if it's too long or so and if it's still too long, chorten the email?

Reply:
This makes sense. A shortened email might not be very useful after all. Let me check with @ciyer if dropping it would work.
I'll still add the check for the `old_active_branch` just in case

Reply:
Then let us drop the email.

Reply:
Ok, I dropped the email and added the logic to limit branch names to 250 chars